### DISCUSSION TITLE
Disable generation of Credentials for the OpenShift Container Registry for service account

### DISCUSSION THREAD

**Discussion by hlavki** [okd-project/okd#1604] [Q&A]
Hi,

is it possible to disable generation of Credentials for the OpenShift Container Registry for concrete service account in OKD?
We are using argoCD to deploy all resources and we need service account with anyuid role binding. When ArgoCD creates new service account, OKD add value to `imagePullSecrets` which argoCD evaluates as not synced.

thanks

**Comment by pflaeging** [comment]
You can set `argocd.argoproj.io/compare-options: IgnoreExtraneous` as Annotation. 

;-)

**Comment by hlavki** [comment]
Hi @pflaeging,

thanks for the suggestion. I already tried to add this annotation to the service account, but it didn't help and I don't want to add this annotation to the whole application.

**Reply by pflaeging** [reply]
Sorry, maybe I was to short:

Yes, you're right, ArgoCD says it's not in sync, but if you set the annotation the app remains in "green" state, though the unsynced parts with the annotation are ignored.

**Reply by hlavki** [reply]
Yes, but if I set the annotation to argo `Application' resource, it will be green for any unsynced resource in this application and I don't want this. The other option (and I think better) is to create the service account outside of argoCD, but I was curious if I can somehow enforce OKD to not create dockercfg secret.  thanks

**Reply by vrutkovs** [reply]
You don't have to mark the whole app - you can mark just the secret, see https://github.com/argoproj/argo-cd/issues/4487#issuecomment-839792534

**Comment by vrutkovs** [comment]
In general - no, its not possible (also SA without generated secrets is probably useless anyway)

**Reply by hlavki** [reply]
Hi @vrutkovs, it's not useless when you add your own dockercfg secret.
